src/librbd/internal.cc

```diff
@@ -727,7 +727,8 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     uint64_t order = *c_order;
 
     ImageOptions opts;
-    opts.set(RBD_IMAGE_OPTION_FEATURES, features);
+    uint64_t new_features = features & ~RBD_FEATURE_DIRTY_CACHE;
```

Comment:
Add `RBD_FEATURE_DIRTY_CACHE` to `RBD_FEATURES_IMPLICIT_ENABLE`

Reply:
Currently DIRTY_CACHE is already in RBD_FEATURES_IMPLICIT_ENABLE. 
In fact, I think the best place to handle RBD_FEATURES_IMPLICIT_ENABLE features is in CloneRequest.cc. 
I am concerned why we need the attribute m_use_p_features 
https://github.com/ceph/ceph/blob/master/src/librbd/image/CloneRequest.cc#L59  
If calling with features, m_use_p_features  is false.
```cpp
  if (m_opts.get(RBD_IMAGE_OPTION_FEATURES, &m_features) == 0) {
    if (m_features & ~RBD_FEATURES_ALL) {
      lderr(m_cct) << "librbd does not support requested features" << dendl;
      complete(-ENOSYS);
      return;
    }
    m_use_p_features = false;
  }
```
It blocks to exclude implicit features: 
```c
 if (m_use_p_features) {
    m_features = (p_features & ~RBD_FEATURES_IMPLICIT_ENABLE);
  }
``` 
https://github.com/ceph/ceph/blob/master/src/librbd/image/CloneRequest.cc#L216-L218